### DISCUSSION TITLE
Warning for static functions called in the same script

### DISCUSSION THREAD

**Discussion by hilfazer** [godotengine/godot-proposals#7447] [Scripting]
The following code 

```
extends Node


static func stafun():
	pass


func _ready() -> void:
	stafun()

```

produces following warning
> The function "stafun()" is a static function but was called from an instance. Instead, it should be directly called from the type: "Node.stafun()"

I understand why (implicit self) but it makes me ask 2 questions:

1. What's the intended way to deal with this warning in this scenario?
2. Should this warning appear in this scenario to begin with?

I've got 1 answer to the first question on Discord which is, as i predicted, to use class_name
```
ClassName.stafun()
```

but it's strange to have a script require a global reference to itself. Moreover scripts without a class_name aren't able to do that.
So perhaps there is some other way (othar than ignoring the warning).

**Comment by dalexeev** [comment]
See also https://github.com/godotengine/godot/issues/77098#issuecomment-1559559143:

> That warning is a bit too eager though, as seen in #74397. So we probably shouldn't take it as design guidance, unless we can make sure it's only emitted in cases where it makes sense.
> 
> It was added mostly as bandaid for issues such as
> 
> ```
> var tex = ImageTexture.new()
> tex.create_from_image(img)
> ```
>
> which were valid code in 3.x but would suddenly do nothing in 4.x, as the method is now static and returns an instance.
> 
> That's arguably the only case which should really trigger a warning (calling a static method on an instance that returns an instance of the same type, i.e. a static constructor).

**Comment by theraot** [comment]
See also: https://github.com/godotengine/godot-proposals/issues/391
